GUI/window.py

```diff
@@ -598,6 +679,11 @@ def change_color(self, binary):
                     self.green_1 = (0,1,0)
         
 
+class ASCIIGrid(Widget):
+
+    def __init__(self, **kwargs):
+        super(ASCIIGrid, self).__init__(**kwargs)
+
```

Comment:
This class is inherently doing nothing, just passing the widget. I think you could move all the labels declared on the run window to this method, then add them to the widget using `self.add_widget`. Then, you could pass the main widget to the run window (like you are already doing) and then add that main widget to the run window. You would also need to change the update method down to that class, and change the calling of the method from `self.update_ascii_grid()` to `self.ascii.update_ascii_grid()`.

Something like this, but with all the labels:

`class ASCIIGrid(Widget):

    def __init__(self, **kwargs):
        super(ASCIIGrid, self).__init__(**kwargs)
        self.ascii_label_1 = Label(text='[color=000000]' + chr(int(RAM[4088], 16)) + '[/color]', pos=(-187, 29),
                                   font_size=40, markup=True)
        self.add_widget(self.ascii_label_1)
    
    def update_ascii_grid(self):
        self.ascii_label_1.text = '[color=000000]' + chr(int(RAM[4088], 16)) + '[/color]'`

Reply:
Was verified and deduced it is alright to leave as is for now.